middleware/index.ts

```diff
@@ -282,6 +284,58 @@ function errorMessage(error: Error): any {
         .build();
 }
 
+function isInitialAsyncRequest(payload: any, headers: any): boolean {
+    return payload.type 
+        && payload.type.startsWith('com.salesforce.function.async')
+        && !headers[ASYNC_HEADER];
+}
+
+// Invoke given function again to fulfill async request; response is not handled
+async function invokeAsyncFn(logger: Logger, payload: any, headers: any): Promise<void> {
+    let hostUrl = headers['Host'] || payload?.data?.sfContext?.resource;
+    if (!hostUrl) {
+        throw new Error('Unable to determine host');
+    }
+
+    headers[ASYNC_HEADER] = 'true';
+    const parsedUrl = require('url').parse(hostUrl);
+    const https = parsedUrl.protocol.startsWith('https');
+
+    const options = {
+        hostname: parsedUrl.hostname,
+        port: parsedUrl.port || (https ? 443 : 80),
+        method: 'POST',
+        path: '/',
+        headers
+    };
+
+    // Invoke function and ignore response
+    const lib = require(https ? 'https' : 'http');
+    await new Promise((resolve, reject) => {
+        const req = lib.request(options);
+        req.on('error', (err) => {
```

Comment:
I can't recall if we said we would try to write back to core here as well or not, but I guess this is where we would do it once we have the core writeback written for the happy path.

Reply:
Yes, good call.  I'll add.